controllers/adapters_controller.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * Adapter Controller.
+ *
+ * Manages HTTP requests to /adapters. 
+ *  
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+var express = require('express');
+var adapterManager = require('../adapter-manager');
+
+var adaptersController = express.Router();
+
+/**
+ * Return a list of adapters
+ */
+adaptersController.get('/', (request, response) => {
+  adapterManager.discoverAdapters().then(function(adapters) {
+    var adapterList = [];
+    for (var adapter of adapters) {
+      adapterList.push(adapter.asDict());
+    }
+    response.json(adapterList);
+  });
+});
+
+/**
+ * Cancel Action: add a new device to the adapter.
+ */
+adaptersController.delete('/addSomeDevice', (request, response) => {
```

Comment:
Oh, so this isn't the API that the front end will call to initiate pairing then? I guess we could implement that how I described...